backend/libs/watcher-operations.spec.ts

```diff
@@ -0,0 +1,275 @@
+                    currentTime: 42,
+                    now: new Date(),
+                };
+                await updateWatcherVideoStatus(
+                    room!,
+                    tableName,
+                    'owner',
+                    playerEvent,
+                    ddb,
+                );
+                await updateWatcherVideoStatus(
+                    room!,
+                    tableName,
+                    'friend',
+                    playerEvent,
+                    ddb,
+                );
+                await getDDBRoom();
+                done();
+            });
+            test.todo(
+                'initiates synchronized start https://github.com/LesGars/WatchWithMe/issues/145',
+            );
+        });
+        describe('when receiving a seek event from the owner', () => {
+            beforeEach(async (done) => {
+                playerEvent = {
+                    mediaEventType: MediaEventType.SEEK,
+                    duration: 42,
+                    currentTime: 48,
+                    now: new Date(),
+                };
+                await updateWatcherVideoStatus(
+                    room!,
+                    tableName,
+                    'owner',
+                    playerEvent,
+                    ddb,
+                );
+                await getDDBRoom();
+                done();
+            });
+            it('updates the owner to the BUFFERING state', async () => {
+                expect(ddbRoom!.watchers['owner']).toEqual({
+                    connectionId: 'owner',
+                    currentVideoStatus: UserVideoStatus.BUFFERING,
+                    id: 'owner',
+                    initialSync: false,
+                    userAgent: 'TODO',
+                });
+            });
+            test.todo(
+                'update room status for https://github.com/LesGars/WatchWithMe/issues/61',
+            );
+        });
+    });
+
+    describe('when the room is in a playing state (synchronized start was initiated previously)', () => {
+        beforeEach(async (done) => {
+            room!.videoStatus = VideoSyncStatus.PLAYING;
+            room!.currentVideoUrl = 'https://www.youtube.com/watch?v=4242';
+            await updateRoom(room!, tableName, ddb);
+            done();
+        });
+        describe('when receiving a play event from the owner', () => {
+            beforeEach(async (done) => {
+                playerEvent = {
+                    mediaEventType: MediaEventType.PLAY,
+                    duration: 42,
+                    currentTime: 42,
+                    now: new Date(),
+                };
+                await updateWatcherVideoStatus(
+                    room!,
+                    tableName,
+                    'owner',
+                    playerEvent,
+                    ddb,
+                );
+                done();
+            });
+            it('updates the watcher to the PLAYING state', async () => {
```

Comment:
It should not happen no ? Or at least it should have no impact no ?

Reply:
This code/test may be obsolete since we currently do not have the appropriate watcher state sent and it's hard to figure out what we should do. We need to validate my terminology proposal in the wiki before I can refactor this.

Reply:
Ok